### PR TITLE
fix(browser): allow updating total specs count

### PR DESCRIPTION
**Pull request by matz3**
This change allows providing the total specs count after the tests have
been started.
The count will be updated in case it has been provided already.
This can be uselful for some adapters where the total specs count can
not be determined before starting the first test.

**Regarding tests**
I wanted to update the tests and/or create a test case but the `onInfo` tests are not clear to me:
- 'browser_dump' doesn't seem to be a valid event
- I'm not sure whether the behavior of the tests is really expected or just works by chance.

So I would be happy to get feedback from you before writing a test case.
https://github.com/karma-runner/karma/blob/c277a6bd130531702e2529f0410aa441328f187e/test/unit/browser.spec.js#L102-L137

### CODE REVIEW

**Comment by johnjbarton** [comment]
> * 'browser_dump' doesn't seem to be a valid event

I guess the test is attempting to cover:
https://github.com/karma-runner/karma/blob/master/lib/browser.js#L63

However it tests something "not happening" and many things do not happen when code is changed. Maybe the event was originally `browser_dump` and was renamed, but the test was not corrected. Please correct it if you can.

File: lib/browser.js [hunk 1]
```diff
@@ -60,6 +60,8 @@ class Browser {
 
     if (helper.isDefined(info.log)) {
       this.emitter.emit('browser_log', this, info.log, info.type)
+    } else if (helper.isDefined(info.total)) {
+      this.lastResult.total = info.total
```
Comment (johnjbarton) [inline lib/browser.js]: If you guard the update to results with state `EXECUTING` then the tests will pass.  You can add a similar test with mock state `EXECUTING` to verify the change.

Comment (matz3) [reply lib/browser.js]: In order to fix the test `should ignore if browser not executing` I would suggest to wrap all those if-clauses with a check for state `EXECUTING`.
Should `this.refreshNoActivityTimeout()` be still called then?

Comment (johnjbarton) [reply lib/browser.js]: I think we should log events in any state, they could be error messages for corner cases.  So remove the line in the test that checks the `.dump` path.

The  `this.refreshNoActivityTimeout()` should always be called, it signals that the browser is still on the line.

Comment (matz3) [reply lib/browser.js]: Makes sense. I've pushed my changes.

**Comment by johnjbarton** [comment]
Thanks!

**Comment by matz3** [comment]
Thanks for the quick handling 👍
